zebrad/src/config.rs

```diff
@@ -54,3 +71,80 @@ pub struct ZebradConfig {
     /// Mining configuration
     pub mining: zebra_rpc::config::mining::Config,
 }
+
+impl ZebradConfig {
+    /// Loads the configuration from the conventional sources.
+    ///
+    /// Configuration is loaded from three sources, in order of precedence:
+    /// 1. Hard-coded defaults (lowest precedence)
+    /// 2. TOML configuration file (if provided)
+    /// 3. Environment variables with `ZEBRA_` prefix (highest precedence)
```

Comment:
It would be nice to have these in the same order as they are in the `generate` cmd output.

```suggestion
    /// 1. Environment variables with `ZEBRA_` prefix (highest precedence)
    /// 2. TOML configuration file (if provided)
    /// 3. Hard-coded defaults (lowest precedence)
```